components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java

```diff
@@ -492,22 +500,9 @@ public static void updatePermissions(String applicationName, ApplicationPermissi
             Collection appNodeCollec = (Collection) tenantGovReg.get(applicationNode);
             String[] childern = appNodeCollec.getChildren();
 
-            // new permissions are null. deleting all permissions case
-            if ((childern != null && childern.length > 0)
-                    && (permissions == null || permissions.length == 0)) { // there are permissions
-                tenantGovReg.delete(applicationNode);
-            }
-
-            if (ArrayUtils.isEmpty(permissions)) {
-                return;
-            }
-
-            // no permission exist for the application, create new
-            if (childern == null || appNodeCollec.getChildCount() < 1) {
-
+            if (childern == null || appNodeCollec.getChildCount() < 1) { // no permissions exist for the application
                 addPermission(applicationNode, permissions, tenantGovReg);
-
-            } else { // there are permission
+            } else { // there are existing permissions for the application
```

Comment:
Lets fix the comment format

Reply:
Outdated. `updatePermissions` method has been refactored in a later change and the comments are no longer applicable.